### PR TITLE
T108: contact form frontend

### PR DESCRIPTION
**Pull request by fahadakhan2**
Screenshot 2023-03-05 at 11 31 09 PM
Screenshot 2023-03-05 at 11 33 51 PM
Screenshot 2023-03-05 at 11 35 05 PM

CSS will be added whenever Ray can get me the figma designs

### CODE REVIEW

**Comment by justinpchen94** [comment]
not a review but if you could add some screenshots, that'd help the reviewers out immensely!

File: frontend/src/components/contacts.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    } catch (error) {
+      console.error(error);
+      setErrorMessage((error as any).response.date ?? "Unknown error occured.");
+    }
+  };
+
+  return (
+    <div className="contacts_form">
+      <h1>Add a contact</h1>
+      <form action="/user/contacts" method="post" onSubmit={handleSubmit}>
```
Comment (kdata12) [inline frontend/src/components/contacts.tsx]: The action and method in not necessary here, the handleSubmit function will take care of your post request!
```suggestion
      <form onSubmit={handleSubmit}>
```

**Comment by chinkevin25** [comment]
if this looks good to @kdata12 i'm good with it too!

File: frontend/src/components/contacts.tsx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        </div>
+        <div>
+          <button className="add_contact_button" type="submit">
+            Add contact
+          </button>
+        </div>
+      </form>
+    </div>
+  );
+};
```
Comment (chinkevin25) [inline frontend/src/components/contacts.tsx:275]: I'm okay w/ merging this if Hyun is okay with it. My only comment is that right now we have free text inputs for event type and relationships and that we're not enforcing a date type to the input (i.e. I can make the date `foo`).

I'm okay with this for now but we may need to revisit this in the future otherwise. 

But good work on getting the dates to sync up! I know that was a little tricky :) 

tldr: form has some "invalid" values that we may want to prevent 
image
image

Comment (fahadakhan2) [reply frontend/src/components/contacts.tsx:275]: Sent some questions about this on discord.

File: frontend/src/components/contacts.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        lastName,
+        relationships,
+        importantDates,
+        email,
+        phoneNumber,
+        notes,
+      });
+    } catch (error) {
+      console.error(error);
+      setErrorMessage((error as any).response.date ?? "Unknown error occured.");
```
Comment (kdata12) [inline frontend/src/components/contacts.tsx]: Misspelled data?
```suggestion
      setErrorMessage((error as any).response.data ?? "Unknown error occured.");
```

File: frontend/src/components/contacts.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+              value={relationshipType}
+              onChange={(event) => {
+                const value = event.target.value;
+                if (/^[A-Za-z\s]*$/.test(value)) {
+                  setRelationshipType(value);
+                }
+              }}
+            />
+          </label>
+          <button onClick={addRelationship}>Add</button>
```
Comment (kdata12) [inline frontend/src/components/contacts.tsx]: The `add` button for relationship and date/event is causing the form to submit which runs the handleSubmit function. We'd want to add a `type="button"` here 
```suggestion
          <button type="button" onClick={addRelationship}>Add</button>
```

File: frontend/src/components/contacts.tsx [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+          {relationships.map((type, index) => (
+            <div key={index}>
+              <input
+                type="text"
+                value={type.type}
+                onChange={(event) =>
+                  handleRelationshipChange(index, "type", event.target.value)
+                }
+              />
+              <button onClick={() => handleRemoveRelationshipType(index)}>
```
Comment (kdata12) [inline frontend/src/components/contacts.tsx]:
```suggestion
              <button type="button" onClick={() => handleRemoveRelationshipType(index)}>
```

File: frontend/src/components/contacts.tsx [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+              value={eventType}
+              onChange={(event) => {
+                const value = event.target.value;
+                if (/^[A-Za-z\s]*$/.test(value)) {
+                  setEventType(value);
+                }
+              }}
+            />
+          </label>
+          <button onClick={addImportantEvent}>Add</button>
```
Comment (kdata12) [inline frontend/src/components/contacts.tsx]:
```suggestion
          <button type="button" onClick={addImportantEvent}>Add</button>
```

File: frontend/src/components/contacts.tsx [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+                }
+              />
+              <input
+                type="text"
+                value={date.event}
+                onChange={(event) =>
+                  handleImportantEventChange(index, "event", event.target.value)
+                }
+              />
+              <button onClick={() => handleRemoveImportantEvent(index)}>
```
Comment (kdata12) [inline frontend/src/components/contacts.tsx]:
```suggestion
              <button type="button" onClick={() => handleRemoveImportantEvent(index)}>
```

File: frontend/src/components/contacts.tsx (hunk 4 above)
Comment (fahadakhan2) [reply frontend/src/components/contacts.tsx:203]: Would this also apply to the remove relationship button? For example
 <button onClick={() => handleRemoveImportantEvent(index)}>
          x
 </button>

Comment (fahadakhan2) [reply frontend/src/components/contacts.tsx]: Good to know will change this!

**Review by kdata12 — APPROVED** [review]
